pkg/apiclient/apiclient.go

```diff
@@ -0,0 +1,76 @@
+package apiclient
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"io"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	pb "go.creack.net/telepilot/api/v1"
+)
+
+// Client is our client business logic. It is the high level API Client.
+type Client struct {
+	conn   *grpc.ClientConn
+	client pb.TelePilotServiceClient
+}
+
+// NewClient returns a client prepared to connect to the server.
+func NewClient(tlsConfig *tls.Config, addr string) (*Client, error) {
+	conn, err := grpc.NewClient(addr, grpc.WithTransportCredentials(credentials.NewTLS(tlsConfig)))
+	if err != nil {
+		return nil, fmt.Errorf("grpc new client: %w", err)
+	}
+	return &Client{conn: conn, client: pb.NewTelePilotServiceClient(conn)}, nil
```

Comment:
In this initial PR it may not amount to much, but having this layer simplifies a lot testing and future improvements/changes as it de-correlate the rest of the code from the actual API. Any change to the API can be handled in this file with little to no impact on the rest of the codebase.